modules/apps/export-import/export-import-service/src/main/java/com/liferay/exportimport/internal/data/handler/BatchEnginePortletDataHandlerRegistrar.java

```diff
@@ -143,32 +185,29 @@ public ServiceRegistration<PortletDataHandler> addingService(
 							"batch.engine.entity.class.name")),
 					_companyLocalService,
 					exportImportVulcanBatchEngineTaskItemDelegate,
-					exportImportDescriptor.getItemClassName(),
+					exportImportDescriptor.getItemClassName(), portletId,
 					(String)serviceReference.getProperty(
 						"batch.engine.task.item.delegate.name"),
 					_userLocalService);
 
 			return _bundleContext.registerService(
 				PortletDataHandler.class, batchEnginePortletDataHandler,
-				HashMapDictionaryBuilder.<String, Object>put(
-					"batch.engine.task.item.delegate.item.class.name",
-					exportImportDescriptor.getItemClassName()
-				).put(
-					"jakarta.portlet.name", portletId
-				).put(
-					"service.ranking", Integer.MAX_VALUE
-				).build());
+				_setEnabledCompanyIds(
```

Comment:
Shouldn't here have a shortcut protection for the _enabledCompanyIds is empty case?

In case there is no company enabling FF LPD-35914, do you really want to register a system scope PortletDataHandler service? 

I am not sure the business logic requirement here. But on the consumer side, you always lookup PortletDataHandler with a companyId, base on this registering a system scope PortletDataHandler should be meaningless, since no one will ever see it.